mesheryctl/internal/cli/root/registry/update.go

```diff
@@ -66,29 +91,41 @@ mesheryctl registry update --spreadsheet-id 1DZHnzxYWOlJ69Oguz4LkRVTFM79kC2tuvdw
 		if err != nil {
 			return ErrUpdateRegistry(err, modelLocation)
 		}
+		utils.Log.Debugf("Logger created and set debug log level at %s", logFilePath)
 		return nil
 	},
 	RunE: func(cmd *cobra.Command, args []string) error {
-
-		srv, err := mutils.NewSheetSRV(spreadsheeetCred)
+		info, err := os.Stat(modelLocation)
 		if err != nil {
-			utils.Log.Error(ErrUpdateRegistry(err, modelLocation))
-			return err
+			return utils.ErrInvalidPath(err, modelLocation)
 		}
-		resp, err := srv.Spreadsheets.Get(spreadsheeetID).Fields().Do()
-		if err != nil || resp.HTTPStatusCode != 200 {
-			utils.Log.Error(ErrUpdateRegistry(err, outputLocation))
-			return err
+		if !info.IsDir() {
+			return utils.ErrInvalidPath(err, modelLocation)
 		}
 
-		sheetGID = GetSheetIDFromTitle(resp, "Components")
+		utils.Log.Debugf("Input Directory check completed with path  %s", modelLocation)
 
-		err = InvokeCompUpdate()
+		var parser ComponentSourceParser
+
+		if csvDir != "" {
+			utils.Log.Info("Using local CSV directory: ", csvDir)
+			parser = &LocalCSVDirParser{dirPath: csvDir}
+		} else {
```

Comment:
```suggestion
		} 
		
		if {
```
best for long term maintenance